crates/datasources/src/clickhouse/convert.rs

```diff
@@ -228,6 +215,78 @@ fn column_to_array(
             }
             Arc::new(vals.finish())
         }
+        DataType::Timestamp(TimeUnit::Second, tz) => {
+            let mut vals = TimestampSecondBuilder::with_capacity(column.len());
+            for val in column {
+                match val {
+                    Value::DateTime(v) => vals.append_value(
+                        DateTime::<Tz>::try_from(v)
+                            .map_err(ClickhouseError::DateTimeConvert)?
+                            .timestamp(),
+                    ),
+                    Value::DateTime64(v) => vals.append_value(
+                        DateTime::<Tz>::try_from(v)
+                            .map_err(ClickhouseError::DateTimeConvert)?
+                            .timestamp(),
+                    ),
+                    Value::Null if nullable => vals.append_null(),
+                    other => {
+                        return Err(ClickhouseError::String(format!(
+                            "unexpected value type: {other}"
+                        )))
+                    }
+                }
+            }
+            Arc::new(vals.finish().with_timezone_opt(tz))
+        }
+        DataType::Timestamp(TimeUnit::Millisecond, tz) => {
+            let mut vals = TimestampMillisecondBuilder::with_capacity(column.len());
+            for val in column {
+                match val {
+                    Value::DateTime(v) => vals.append_value(
+                        DateTime::<Tz>::try_from(v)
+                            .map_err(ClickhouseError::DateTimeConvert)?
+                            .timestamp_millis(),
+                    ),
+                    Value::DateTime64(v) => vals.append_value(
+                        DateTime::<Tz>::try_from(v)
+                            .map_err(ClickhouseError::DateTimeConvert)?
+                            .timestamp_millis(),
+                    ),
+                    Value::Null if nullable => vals.append_null(),
+                    other => {
+                        return Err(ClickhouseError::String(format!(
+                            "unexpected value type: {other}"
+                        )))
+                    }
+                }
+            }
+            Arc::new(vals.finish().with_timezone_opt(tz))
+        }
+        DataType::Timestamp(TimeUnit::Microsecond, tz) => {
+            let mut vals = TimestampMicrosecondBuilder::with_capacity(column.len());
+            for val in column {
+                match val {
+                    Value::DateTime(v) => vals.append_value(
+                        DateTime::<Tz>::try_from(v)
```

Comment:
do these conversions work across resolutions too? maybe good to have a comment about their rounding/translation strategy.

Reply:
What do you mean by resolutions?

Reply:
I think what I'm asking is "what are the cases where the `try_from` fails? Is it out-of-bounds issues only?

Reply:
Yup, it's only out-of-bound cases.